### ISSUE TITLE
Rviz crush due to addition of jsk recognition message

### ISSUE THREAD

**Issue by tiandaji** [tigerk0430/voxelnet_ros#1]
Hello, @tigerk0430 !
I am interested in your voxelnet implementation in ROS. However, I've encountered the rviz crush whenever I tried to visualize jsk recognition box.
I was able to compile the source code via catkin_make,  by running tensorflow and python3, the voxelnet_ros.py node worked as well while testing my own bagfile. I was sure about this because the node had this kind of output, which means the recognition was done:

arr_bbox.boxes.size() : 5 
[+] [x,y,z,dx,dy,dz] : 17.684256, -5.5233808, -0.6701406, 3.776594, 1.5653551, 1.4613458
[+] [x,y,z,dx,dy,dz] : 28.313154, -2.416056, -0.5163301, 4.0180697, 1.6209732, 1.4673424
[+] [x,y,z,dx,dy,dz] : 23.92562, 0.52841944, -1.0636827, 3.6418939, 1.5527209, 1.4772903
[+] [x,y,z,dx,dy,dz] : 12.590783, 10.839781, -1.463708, 4.003292, 1.6194181, 1.4821359
[+] [x,y,z,dx,dy,dz] : 21.749434, 13.669043, -1.6811992, 4.125592, 1.6347773, 1.5180384
[+] FIRST================
[+] tmp_boxes3d: (289, 7), tmp_scores: (289,)
[+] SECOND================
[+] tmp_boxes3d: (5, 7), tmp_scores: (5,)
[+] ===============FINISHIED================

For the jsk dependency and rviz plugins part, I have installed 
sudo apt-get install ros-kinetic-jsk-hark-msgs
sudo apt-get install ros-kinetic-people-msgs
sudo apt-get install ros-kinetic-view-controller-msgs
in order to solve compilation issue;
Then installed the jsk-visualization pkg

sudo apt-get install ros-kinetic-jsk-visualization
The following additional packages will be installed:
  python-pycurl python-urlgrabber ros-kinetic-jsk-interactive ros-kinetic-jsk-interactive-marker
  ros-kinetic-jsk-interactive-test ros-kinetic-jsk-rqt-plugins **ros-kinetic-jsk-rviz-plugins**
Suggested packages:
  libcurl4-gnutls-dev python-pycurl-dbg python-pycurl-doc
The following NEW packages will be installed:
  python-pycurl python-urlgrabber ros-kinetic-jsk-interactive ros-kinetic-jsk-interactive-marker
  ros-kinetic-jsk-interactive-test ros-kinetic-jsk-rqt-plugins ros-kinetic-jsk-rviz-plugins
  ros-kinetic-jsk-visualization

We see all the topic and the topic info:

/clicked_point
/clock
/imu/data
/initialpose
/move_base_simple/goal
/rosout
/rosout_agg
/tf
/tf_static
/velodyne_points
/velodyne_points_modified
/voxelnet_arr_bbox
/voxelnet_bbox

Type: jsk_recognition_msgs/BoundingBoxArray

Publishers: 
 * /voxelnet_ros_node (http://daji-laptop:44817/)

Subscribers: None

And the plugins are there in rviz:
![screenshot from 2018-10-15 15-14-22](https://user-images.githubusercontent.com/35362216/46935362-8713fb80-d08d-11e8-99d1-fa2b181c7a1a.png)
As soon as I select the BoundingBox and BoundingBoxArray in jsk_rviz_plugins, the rviz crushes.
Do you have any idea about this?
**One more thing**, since you used the voxelnet generated by qianguih https://github.com/qianguih/voxelnet
I suppose you may help me with its training and validation dataset issue: https://github.com/qianguih/voxelnet/issues/22
Thanks very much in advance.

**Comment by gyubeomim** [comment]
Hello @tiandaji 

Thank you for the posting first issue!

first of all,
I haven't had this kind of issue and I wonder what kind of error meesage you've got as soon as you crushed.

I installed all kind of jsk packages like 
`sudo apt-get install ros-kinetic-jsk*`

and error message wans't occured. 
try this out and give me a feedback and post an error message if not working.

and Thank you for https://github.com/qianguih/voxelnet/issues/22  :-)

**Comment by tiandaji** [comment]
Hello @tigerk0430 
Thanks for your response.
I've tried to install all the jsk pkgs this morning by typing
sudo apt-get install ros-kinetic-jsk*
However, there was no change for the crush issue: just like I said in the previous post, everything was normal, but once I chose to add the boundbox topic, rviz UI disappeared instantly, leaving only "segment fault" in the terminal where I had opened the rivz.
As for the issue 22 in the voxelnet, I was asking about the data structure qianguih/voxelnet#22 in the third post, because I could not find the validation set for label_2.
Thank you very much for any help, as I am really stuck for days.

**Comment by tiandaji** [comment]
Hello @tigerk0430 
Thanks for your response.
I've tried to install all the jsk pkgs this morning by typing
sudo apt-get install ros-kinetic-jsk*
However, there was no change for the crush issue: just like I said in the previous post, everything was normal, but once I chose to add the boundbox topic, rviz UI disappeared instantly, leaving only "segment fault" in the terminal where I had opened the rivz.
As for the issue 22 in the voxelnet, I was asking about the data structure qianguih/voxelnet#22 in the third post, because I could not find the validation set for label_2.
Thank you very much for any help, as I am really stuck for days.

**Comment by tiandaji** [comment]
Hello @tigerk0430 
Sorry for the Internet connection issue, maybe I posted this answer several times.
Thanks for your response.
I've tried to install all the jsk pkgs this morning by typing
sudo apt-get install ros-kinetic-jsk*
However, there was no change for the crush issue: just like I said in the previous post, everything was normal, but once I chose to add the boundbox topic, rviz UI disappeared instantly, leaving only "**segment fault**" in the terminal where I had opened the rivz. Or maybe it is due to some bag problem? Would you like to share your **bagfile**?
As for the issue 22 in the voxelnet, I was asking about the data structure qianguih/voxelnet#22 in the third post, because I could not find the **validation set for label_2**. As you are using voxelnet, I suppose you have the experience to train and validate the model through KITTI dataset.
Thank you very much for any help, as I am really stuck for days.

**Comment by tiandaji** [comment]
Hello @tigerk0430 
Thanks for your response.
I've tried to install all the jsk pkgs this morning by typing
sudo apt-get install ros-kinetic-jsk*
However, there was no change for the crush issue: just like I said in the previous post, everything was normal, but once I chose to add the boundbox topic, rviz UI disappeared instantly, leaving only "segment fault" in the terminal where I had opened the rivz. Or maybe it is due to some bag problem? Would you like to share your bagfile?
As for the issue 22 in the voxelnet, I was asking about the data structure qianguih/voxelnet#22 in the third post, because I could not find the validation set for label_2. If you are using voxelnet, I suppose you have the experience to train and validate the model through KITTI dataset.
Thank you very much for any help, as I am really stuck for days.

**Comment by gyubeomim** [comment]
@tiandaji 

oh, that's weird :-/ I haven't had this kind of SegFault. 
ok let me share temporal rosbag file using V-Rep Vehicle Simulator
http://gofile.me/3TjkG/pRaHzhNHe

I've tested this rosbag file with voxelnet_ros. result looks like below
![peek 2018-10-25 17-01](https://user-images.githubusercontent.com/18214498/47486326-103fe500-d87b-11e8-912f-63d036cebfea.gif)

**Comment by tiandaji** [comment]
thx for your response @tigerk0430 , but when I saw the notification and began the download, the file was already expired... you may share your file (not necessarily large, several dozens of Megabytes, lasting for 10~20 seconds, are sufficient ) via dropbox or googledrive, and send me the link to tiandaji@gmail.com if you would not like it to go public.
thx again in advance.

**Comment by gyubeomim** [comment]
@tiandaji 

oops my bad. 
I'll upload it on Google Drive instead.

https://drive.google.com/drive/folders/1yKPUnWpqE28gBTYDIOPekJfwmZEc0wBN?usp=sharing


If It doesn't work again, I'll send it to your email

**Comment by tiandaji** [comment]
hi @tigerk0430 
The bag is good to download, and will you please tell me the frame name of velodyne_points in the world reference in order that I could visualize the result?
Thx so much.

**Comment by gyubeomim** [comment]
hello @tiandaji 

the frame_id of velodyne is `velodyne_link` which you can find from this script Line 104
![01](https://user-images.githubusercontent.com/18214498/47640752-ad668a80-dba7-11e8-80b2-61c69849d8f0.png)

if you set frame_id of RViz to `map` or `base_link`, you can visualize Velodyne data as well :-)

**Comment by irmakguzey-dof** [comment]
Hello there, 
I am having a similar problem but this time with jsk_recognition_msg/PolygonArray. 
When I run a script that appends more Polygons to this message according to the published points on rviz, and show this PolygonArray using rviz plugin, rviz immediately crushes without even giving any error message other than giving:

`[rviz-4] process has died [pid 22924, exit code -11, cmd /opt/ros/melodic/lib/rviz/rviz -d /home/irmak/Workspace/ros_ws/src/dof-robotics-kamil/kamil_navigation/rviz/kamil_navigation.rviz __name:=rviz __log:=/home/irmak/.ros/log/c38a4c6a-7697-11eb-989a-c809a8c5b962/rviz-4.log].
log file: /home/irmak/.ros/log/c38a4c6a-7697-11eb-989a-c809a8c5b962/rviz-4*.log`

When I look at the .log files, they are empty.
I have no idea why this is happening. When I try to visualize the last element of this PolygonArray with using Polygon, it works fine. 
It would be great if you could help me a little. 
Thanks a lot!
